client/serviceinterface_update.go

```diff
@@ -267,8 +274,8 @@ func removeServiceInterfaceTarget(serviceName string, targetName string, deleteI
 		if err != nil {
 			return fmt.Errorf("Failed to read json for service interface %s: %s", serviceName, err)
 		}
-		if service.IsAnnotated() && kube.IsOriginalServiceModified(service.Address, cli.Namespace, cli.GetKubeClient()) {
-			_, err = kube.RemoveServiceAnnotations(service.Address, cli.Namespace, cli.KubeClient, []string{types.ProxyQualifier})
+		if service.IsAnnotated() && kube.IsOriginalServiceModified(service.Address, namespace, cli.GetKubeClient()) {
```

Comment:
In the non-annotated case, it is ignoring the target namespace,
and the `unbind` operation is executed regardless of the namespace provided.

Reply:
I see your point @fgiorgetti... I've just come to think, we might want to expose from more than 2, maybe 5 or 10? Could this be an issue, maybe performance-wise?

Reply:
Today it is already allowed to bind a service to multiple targets, so for now I believe it should be fine
to keep it without a predefined limit.

Reply:
Hey @fgiorgetti. If I didn't miss anything it should be fixed now. When binding or unbinding a target, it should solely do it for a specific namespace. If the exposed service hasn't been created previously with a namespace, it will only consider the target name. Moreover, I also tested and hopefully covered that the requests should be correctly distributed among targets in different namespaces.

Still on this matter, I've noticed one more point regarding this cross-namespace exposure of resources. Taking a quick look at the definition_monitor.go more carefully, I've come to realize that this change won't contemplate the annotated resources since the current informers only watch resources on the namespace where skupper installed. Not sure if this is an issue since before this PR we already could expose "services" in different namespaces (and those were also not covered).

Reply:
Thanks @iagodvsantos. I will review and share my thoughts.

With regards to the annotated issue, I believe it can be handled as part of a separate PR.